sources/packages/web/src/components/common/modals/EducationProgramDeactivationModal.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <v-form ref="deactivateProgramForm">
+    <modal-dialog-base
+      :showDialog="showDialog"
+      title="Confirm deactivation"
+      ref="confirmDeactivationModal"
+      okLabel="Confirm deactivation"
+      keepModalOpen="true"
+      ><template #content
+        ><span class="font-bold"
+          >Please confirm you no longer require this program?</span
+        >
+        <p class="mt-5">
+          Confirming deactivation means that this will no longer be visible to
+          students when starting an application. It wil not impact any students
+          with active applications for offerings under this program. This action
+          cannot be undone.
+        </p>
+        <v-textarea
+          v-if="notesRequired"
+          v-model="noteDescription"
+          variant="outlined"
+          label="Note"
+          :rules="[checkNotesLengthRule]"
+          required
+          class="mt-4"
+          hide-details="auto"
+        ></v-textarea>
+      </template>
+      <template #footer>
+        <footer-buttons
+          primaryLabel="Confirm deactivation"
+          primaryButtonColor="danger"
+          @primaryClick="submit"
+          @secondaryClick="cancel"
+          :processing="loading"
+        />
+      </template>
+    </modal-dialog-base>
+  </v-form>
+</template>
+<script lang="ts">
+import { ref, defineComponent } from "vue";
+import { useModalDialog, useRules } from "@/composables";
+import { VForm } from "@/types";
+import { DeactivateProgramAPIInDTO } from "@/services/http/dto";
+
+export default defineComponent({
+  props: {
+    notesRequired: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  setup() {
+    const { showDialog, resolvePromise, showModal, loading, hideModal } =
+      useModalDialog<DeactivateProgramAPIInDTO | boolean>();
+    const { checkNotesLengthRule } = useRules();
+    const noteDescription = ref("");
+    const deactivateProgramForm = ref({} as VForm);
+
+    const submit = async () => {
+      const validationResult = await deactivateProgramForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      const payload = {
+        note: noteDescription.value,
+      };
+      await resolvePromise(payload);
```

Comment:
Should we await on resolvePromise to reset the form?? Is it to allow the user to submit the same information again when the promise is rejected??

Reply:
Yes, waiting will keep the form's current state and allow it to be submitted again.